tests/automation/disappearing_message_checks.spec.ts

```diff
@@ -0,0 +1,305 @@
+import {
+  sendLinkPreview,
+  sendMedia,
+  sendVoiceMessage,
+  trustUser,
+} from './utilities/send_media';
+import { setDisappearingMessages } from './utilities/set_disappearing_messages';
+import {
+  clickOnElement,
+  clickOnTestIdWithText,
+  hasElementBeenDeleted,
+  hasTextMessageBeenDeleted,
+  typeIntoInput,
+  waitForElement,
+  waitForLoadingAnimationToFinish,
+  waitForTestIdWithText,
+  waitForTextMessage,
+} from './utilities/utils';
+import { longText } from './constants/variables';
+import { waitForSentTick } from './utilities/message';
+import { joinCommunity } from './utilities/join_community';
+import { testCommunityName } from './constants/community';
+
+// Disappearing time settings for all tests
+const timeOption: DMTimeOption = 'time-option-30-seconds';
+const disappearingMessageType = 'disappear-after-send-option';
+const disappearAction = 'sent';
+
+[
+  {
+    mediaType: 'image',
+    path: 'fixtures/test-image.png',
+    attachmentType: 'media',
+  },
+  {
+    mediaType: 'video',
+    path: 'fixtures/test-video.mp4',
+    attachmentType: 'media',
+  },
+  {
+    mediaType: 'gif',
+    path: 'fixtures/test-gif.gif',
+    attachmentType: 'media',
+  },
+  {
+    mediaType: 'document',
+    path: 'fixtures/test-file.pdf',
+    attachmentType: 'file',
+  },
+  {
+    mediaType: 'voice',
+    path: '',
+    attachmentType: 'audio',
+  },
+].forEach(({ mediaType, path, attachmentType }) => {
+  test_Alice_1W_Bob_1W(
+    `Send disappearing ${mediaType} 1:1`,
+    async ({ alice, aliceWindow1, bob, bobWindow1 }) => {
+      const testMessage = `${alice.userName} sending disappearing ${mediaType} to ${bob.userName}`;
+      await createContact(aliceWindow1, bobWindow1, alice, bob);
+      // Set disappearing messages
+      await setDisappearingMessages(
+        aliceWindow1,
+        ['1:1', disappearingMessageType, timeOption, disappearAction],
+        bobWindow1,
+      );
+      await Promise.all([
+        waitForTestIdWithText(
+          aliceWindow1,
+          'disappear-control-message',
+          englishStrippedStr('disappearingMessagesSetYou')
+            .withArgs({
+              time: '30 seconds',
+              disappearing_messages_type: disappearAction,
+            })
+            .toString(),
+        ),
+        waitForTestIdWithText(
+          bobWindow1,
+          'disappear-control-message',
+          englishStrippedStr('disappearingMessagesSet')
+            .withArgs({
+              time: '30 seconds',
+              disappearing_messages_type: disappearAction,
+              name: alice.userName,
+            })
+            .toString(),
+        ),
+      ]);
+      // Send media
+      if (mediaType === 'voice') {
+        await sendVoiceMessage(aliceWindow1);
+      } else {
+        await sendMedia(aliceWindow1, path, testMessage);
+      }
+      // Click on untrusted attachment
+      await trustUser(bobWindow1, alice.userName, attachmentType as MediaType);
```

Comment:
if you make the `attachmentType: 'document'` as const (and the others too), you should be able to get rid of that as MediaType. Let me know if not.